chat-plugins/modlog.js

```diff
@@ -278,22 +327,40 @@ exports.commands = {
 		let roomId = (room.id === 'staff' ? 'global' : room.id);
 		const hideIps = !user.can('lock');
 
+		let targetRoom = room;
+
 		if (target.includes(',')) {
 			let targets = target.split(',');
-			target = targets[1].trim();
+			target = targets.slice(1).join(',').trim();
 			roomId = toId(targets[0]) || room.id;
-		}
 
-		let targetRoom = Rooms.search(roomId);
-		// if a room alias was used, replace alias with actual id
-		if (targetRoom) roomId = targetRoom.id;
-		if (roomId.startsWith('battle-') || roomId.startsWith('groupchat-')) return this.errorReply("Battles and groupchats don't have individual modlogs.");
+			// determine permissions, then test if the user 
+			if (['all', 'public'].includes(roomId)) {
+				if (!user.can('modlog')) {
+					if (!this.can('modlog', null, targetRoom)) return; // actual permission check with current room
```

Comment:
This permission checking looks really weird.

What was wrong with how it was before?

Reply:
This is the result I had after talking with @asgdf - it does a rather complex permission checking

1) Checks if the user is allowed to use /modlog in the scope specified (all, public, specified room)
2) If not, include the stuff before the first comma in the modlog query is included into the search, and the room defaults back to where the command is used, and the permission check for the current room is used.

It's basically something that shifts your search (room: 'all', query: 'abc"') => (room: current, query: '"all, abc"') if you do not have permission to do the search in the "specified" room.

Overall, it does not affect how modlog works in general, except for not getting an "Access denied" when you have a comma in a "same-room" search.

Reply:
Eh I'm still not really a fan of intrepreting arguments in a different way based on context information. If I try to check a private room's modlog (which I can't do but keep forgetting about), I'd very much prefer getting an error rather than a *wrong* result.
The options are kinda limited, but overall I still think I'd prefer the AND/OR or &&/|| over this.

Reply:
At the moment we have self-interpreting arguments, as well as the javascript logic symbols.  Is having only the logic operators better than both?